src/Arcus.Templates.Tests.Integration/Authentication/v1/SharedAccessKeyAuthenticationOptionTests.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Arcus.Templates.Tests.Integration.Fixture;
+using Xunit;
+using Xunit.Abstractions;
+using static Arcus.Templates.Tests.Integration.Fixture.InMemorySecretProvider;
+
+namespace Arcus.Templates.Tests.Integration.Authentication.v1
+{
+    [Collection("Integration")]
+    public class SharedAccessKeyAuthenticationOptionTests
+    {
+        private readonly TestConfig _configuration;
+        private readonly ITestOutputHelper _outputWriter;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SharedAccessKeyAuthenticationOptionTests"/> class.
+        /// </summary>
+        public SharedAccessKeyAuthenticationOptionTests(ITestOutputHelper outputWriter)
+        {
+            _configuration = TestConfig.Create();
+            _outputWriter = outputWriter;
+        }
+
+        [Fact]
+        public async Task SharedAccessKeyAuthenticationOption_GetHealthUnauthenticated_ResultsUnauthorized()
+        {
+            // Arrange
+            const string headerName = "x-shared-access-key";
+            const string secretKey = "MySecretKey";
+            string secretValue = Guid.NewGuid().ToString("N");
+
+            var unauthenticatedArguments = 
+                new WebApiProjectOptions()
+                    .WithSharedAccessAuthentication(headerName, secretKey, secretValue);
+            
+            using (var project = await WebApiProject.StartNewAsync(_configuration, unauthenticatedArguments, _outputWriter))
+            {
+                // Act
+                using (HttpResponseMessage response = await project.Health.GetAsync())
+                {
+                    // Assert
+                    Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task SharedAccessKeyAuthenticationOption_GetHealthAuthenticated_ResultsOk()
```

Comment:
Example of how you can test shared access key authentication option:
1. provide generated secret
2. create project options model
3. pass along to web project (which will create a temp project)
4. access endpoints on the created project to see the added authentication